pkg/artifactory/resource_artifactory_virtual_rpm_repository.go

```diff
@@ -0,0 +1,59 @@
+package artifactory
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+var rpmVirtualSchema = mergeSchema(baseVirtualRepoSchema, map[string]*schema.Schema{
+
+	"primary_keypair_ref": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
+		Description:      "Primary keypair used to sign artifacts.",
+	},
+	"secondary_keypair_ref": {
+		Type:             schema.TypeString,
+		Optional:         true,
+		ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
+		Description:      "Secondary keypair used to sign artifacts.",
+	},
+})
+
+type CommonRpmDebianVirtualRepositoryParams struct {
+	PrimaryKeyPairRef   string `hcl:"primary_keypair_ref" json:"primaryKeyPairRef,omitempty"`
+	SecondaryKeyPairRef string `hcl:"primary_keypair_ref" json:"secondaryKeyPairRef,omitempty"`
```

Comment:
Same question (always), do we need `omitempty`? I doubt we need it so let's always start without.

Reply:
Please assume you don't need it. Also, is there a reason no one is using the built in functionality to general the hcl and then using a tag? Usage of `hcl:` was only needed before I implemented auto-detect. If you add it, it's an override.